### PR TITLE
Backend/elasticsearch implementation

### PR DESCRIPTION
**Pull request by kbrabrand**

### CODE REVIEW

**Comment by kbrabrand** [comment]
I messed up when rebasing with remote master, so some of the commits show twice..

**Comment by kbrabrand** [comment]
@fangel: Please take a look and merge the PR. I would like to start writing some tests, but in order to avoid conflicts I'll let you add the PHPUnit bootstrap and such.

File: config/config.dist.php [hunk 1]
```diff
@@ -1,10 +1,13 @@
 <?php
+
+namespace Imbo\MetadataSearch;
+
 return [
     'eventListeners' => [
         'metadata' => [
             'listener' => 'Imbo\MetadataSearch\EventListener\MetadataOperations',
```
Comment (fangel) [inline config/config.dist.php]: Shouldn't we be consistant here? You removed the Imbo\MetadataSearch prefix from params, but not from the listener-name - is this intentional?

Comment (kbrabrand) [inline config/config.dist.php]: When providing the listener class name as a string Imbo expects it to be a FQCN. We could provide it as an instance like we do with the backend, but I'm not sure if that's better or worse.

Comment (kbrabrand) [inline config/config.dist.php]: I'm torn. Is this better?

```php
return [
    'eventListeners' => [
        'metadata' => [
            'listener' => new EventListener\MetadataOperations([
                'backend' => new Backend\ElasticSearch(new \Elasticsearch\Client())
            ])
        ],
    ],
];
```

Comment (fangel) [inline config/config.dist.php]: Oh, right - I didn't spot that it was a string. Suits me right for trying to reviews while eating breakfast :)

Comment (kbrabrand) [inline config/config.dist.php]: :)

Comment (fangel) [inline config/config.dist.php]: Hmm. I think I ever-so-slightly prefer the alternative (`new EventListener\...`). But I'd be okay with either.

Comment (kbrabrand) [inline config/config.dist.php]: I'll go with the alternative then.

File: library/EventListener/MetadataOperations.php [hunk 2]
```diff
@@ -6,6 +6,7 @@
 use Imbo\EventManager\EventInterface;
 use Imbo\Exception\InvalidArgumentException;
 use Imbo\MetadataSearch\Interfaces\SearchBackendInterface;
+use Imbo\Model;
```
Comment (fangel) [inline library/EventListener/MetadataOperations.php]: Why is this `use Imbo\Model` added? It doesn't seem to be used anywhere...

Comment (kbrabrand) [inline library/EventListener/MetadataOperations.php]: Left over from some earlier testing. Removed now.

**Comment by fangel** [comment]
Looks good to me. Merging and closing.

**Pull request merged**